service/discovery/aws/aws.go

```diff
@@ -39,20 +42,52 @@ var log = logrus.WithField("component", "aws-discovery")
 // loadDefaultConfig holds config.LoadDefaultConfig() so the test function can mock it
 var loadDefaultConfig = config.LoadDefaultConfig
 
+// newFromConfigSTS holds sts.NewFromConfig() so the test function can mock it
+var newFromConfigSTS = loadSTSClient
+
 // Client holds configurations across all services within AWS
 // TODO(lebogg): deepsource.io wants the struct to exported since NewAwsStorageDiscovery is exported. Encapsulation?
 type Client struct {
 	Cfg aws.Config
+	// accountID is needed for ARN creation
+	accountID *string
+}
+
+// STSAPI describes the STS api interface (for mock testing)
+type STSAPI interface {
+	GetCallerIdentity(ctx context.Context, params *sts.GetCallerIdentityInput, optFns ...func(*sts.Options)) (*sts.GetCallerIdentityOutput, error)
 }
 
 // NewClient constructs a new AwsClient
 // TODO(lebogg): "Overload" (switch) with staticCredentialsProvider
 func NewClient() (*Client, error) {
 	c := &Client{}
+
+	// load configuration
 	cfg, err := loadDefaultConfig(context.TODO())
 	if err != nil {
-		log.Errorf("Could not load default config: %v", err)
+		return c, err
 	}
 	c.Cfg = cfg
+
+	// load accountID
+	stsClient := newFromConfigSTS(cfg)
+	resp, err := stsClient.GetCallerIdentity(context.TODO(), &sts.GetCallerIdentityInput{})
+	if err != nil {
+		return c, err
```

Comment:
```suggestion
		return nil, fmt.Errorf("could not get caller identity: %w", err)
```

Reply:
Since it is an error within an AWS service call, I will wrap it with AWS own  smithy.APIError interface type.